internal/cli/auth/register.go

```diff
@@ -113,14 +122,33 @@ func (opts *RegisterOpts) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	_, _ = fmt.Fprintf(opts.login.OutWriter, "Successfully logged in as %s.\n", s)
+	_, _ = fmt.Fprintf(opts.OutWriter, "Successfully logged in as %s.\n", s)
 	if opts.login.SkipConfig {
 		return opts.login.config.Save()
 	}
 
 	return nil
 }
 
+func (opts *RegisterOpts) PreRun(cmd *cobra.Command) error {
+	opts.OutWriter = cmd.OutOrStdout()
+	opts.login.OutWriter = cmd.OutOrStdout()
+	opts.login.config = config.Default()
+	if config.OpsManagerURL() != "" {
+		opts.login.OpsManagerURL = config.OpsManagerURL()
+	}
+	return opts.login.initFlow()
+}
+
+func (opts *RegisterOpts) registerPreRun() error {
+	if hasUserProgrammaticKeys() {
+		return fmt.Errorf(`you have already set the programmatic keys for this profile. 
+
+Run '%s auth register --profile <profileName>' to use your username and password with a new profile`, config.BinName())
+	}
+	return nil
+}
```

Comment:
should we have a const alreadyRegisteredError = fmt.Errorf(`you have already ....) then in setup_cmd, check if the opts.register.PreRun(cmd) returns this err (which from `setup` point of view is not an error, the flow should continue)?

Reply:
we'll not display this error for setup, that is why I've separated this from the PreRun. This is not being used on setup.

I'll send you the link for the wording in private.